doc/cross-compilation.xml

```diff
@@ -6,17 +6,17 @@
   <title>Introduction</title>
 
   <para>
-   "Cross-compilation" means compiling a program on one machine for another
-   type of machine. For example, a typical use of cross compilation is to
-   compile programs for embedded devices. These devices often don't have the
-   computing power and memory to compile their own programs. One might think
-   that cross-compilation is a fairly niche concern, but there are advantages
-   to being rigorous about distinguishing build-time vs run-time environments
-   even when one is developing and deploying on the same machine. Nixpkgs is
-   increasingly adopting the opinion that packages should be written with
-   cross-compilation in mind, and nixpkgs should evaluate in a similar way (by
-   minimizing cross-compilation-specific special cases) whether or not one is
-   cross-compiling.
+   "Cross-compilation" means compiling a program on one machine for another type
+   of machine. For example, a typical use of cross-compilation is to compile
+   programs for embedded devices. These devices often don't have the computing
+   power and memory to compile their own programs. One might think that
+   cross-compilation is a fairly niche concern. However, there are significant
+   advantages to rigorously distinguishing between build-time and run-time
+   environments! This applies even when one is developing and deploying on the
```

Comment:
Since this is an intro, I was going to leave the specifics out. We can put it somewhere else though. Basically:

- Make sure nativeBuildInputs don't end up in the package they built (via disallowedReferences)
  - In addition we can avoid patching shebangs of things that shouldn't actually be needed at runtime.
- Make sure a package doesn't use buildInputs at build time (via PATH/HOST_PATH)
- With #48803, we can also use cross infrastructure to only override packages that are being built - not what is used to build them. This can be useful for lots of things. For instance, I want to use x version of bash with a package but don't want to rebuild my entire stdenv.